packages/core/src/dialects/mysql/query-generator-typescript.ts

```diff
@@ -29,6 +31,42 @@ export class MySqlQueryGeneratorTypeScript extends AbstractQueryGenerator {
     return `SHOW FULL COLUMNS FROM ${this.quoteTable(tableName)};`;
   }
 
+  addConstraintQuery(tableName: TableNameOrModel, options: AddConstraintQueryOptions) {
+    return joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(tableName),
+      'ADD',
+      this.getConstraintSnippet(tableName, options),
+    ]);
+  }
+
+  showConstraintsQuery(tableName: TableNameOrModel, constraintName?: string) {
+    const table = this.extractTableDetails(tableName);
+
+    return joinSQLFragments([
```

Comment:
I'm realizing a lot of this code is identical to the "get foreign key constraints" query, which makes sense, they're constraints. I think we may be able to unify their code, maybe?

Reply:
The easiest method would be to add a `constraintType` option to the `ShowConstraintsQueryOptions` which accepts one a constraint type (i.e. FOREIGN KEY) and then add this as part of the where clause in the SQL statement.

This option would also remove the need for `getPrimaryKeyConstraintQuery` and `getDefaultConstraintQuery` method in the mssql dialect as well as `getForeignKeyQuery` in all dialects.

Essentially, we could use one method to return a consistent result across all constraint types.

In a follow-up PR I can look at migrating the `getForeignKeyReferencesForTable` and `getForeignKeysForTables` methods to TS which should sort out any bugs.

Reply:
I think this would be a good idea, but definitely for another PR